rasa/server.py

```diff
@@ -1557,33 +1569,58 @@ def _validate_json_training_payload(rjs: Dict) -> None:
             docs=_docs("/api/http-api"),
         )
 
-
 def _training_payload_from_yaml(
     request: Request, temp_dir: Path, file_name: Text = "data.yml"
 ) -> Dict[Text, Any]:
     logger.debug("Extracting YAML training data from request body.")
 
     decoded = request.body.decode(rasa.shared.utils.io.DEFAULT_ENCODING)
     _validate_yaml_training_payload(decoded)
+    request_payload = rasa.shared.utils.io.read_yaml(
+            decoded, reader_type=["safe"]
+        )
+    
+    if "nlu" in request_payload:
+        nlu_path = os.path.join(temp_dir, API_TEMP_DATA_FOLDER, API_TEMP_NLU_YAML_FILE)
+        nlu_data = {'nlu': request_payload["nlu"]}
+        rasa.shared.utils.io.write_yaml(nlu_data, nlu_path)
```

Comment:
Thank you @ancalita ! Great suggestion! 
P.S. The reason that wrote these lines repeatedly in the beginning is because of taking each 5 keys as independent processes and not sure whether needing to deal with subkeys under them respectively. With above assumption, easily debugging in each process under the single request out of loop is also be taken into consideration. Whatever, it's great news that there's no need further processing~ 😄!